### PR TITLE
Allow NuGet within the IDE

### PR DESCRIPTION
**Pull request by sharwell**
As far as I can tell in testing, this change removes the requirement that users run **Restore.cmd** manually before opening **Roslyn.sln**.

### CODE REVIEW

**Comment by RikkiGibson** [comment]
any evidence one way or the other about the perf impact of this?

**Comment by sharwell** [comment]
@RikkiGibson I tried it for a while and it seems much better than in the past.

**Comment by sharwell** [comment]
@tmat @jaredpar are you OK with this change as well?

File: eng/targets/GenerateCompilerExecutableBindingRedirects.targets [hunk 1]
```diff
@@ -14,7 +14,7 @@
           DependsOnTargets="ResolveAssemblyReferences"
           Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework'">
     <ItemGroup>
-      <SuggestedBindingRedirects Include="@(ReferencePath->'%(FusionName)')" MaxVersion="%(ReferencePath.Version)" KeepMetadata="-none-" Condition="'%(ReferencePath.FrameworkFile)' != 'true'" />
+      <SuggestedBindingRedirects Include="@(ReferencePath->'%(FusionName)')" MaxVersion="%(ReferencePath.Version)" KeepMetadata="-none-" Condition="'%(ReferencePath.FrameworkFile)' != 'true' AND '%(ReferencePath.Version)' != ''" />
```
Comment (tmat) [inline eng/targets/GenerateCompilerExecutableBindingRedirects.targets]: Nit: lower case `AND`

Comment (tmat) [reply eng/targets/GenerateCompilerExecutableBindingRedirects.targets]: I'd also add a comment saying that the Version is not available in design time builds, just so if anyone wonders why the condition is in place.

**Comment by jaredpar** [comment]
I'm on board with this change. The only reason was performance. If that's been addressed then no reason to keep this disabled.

**Comment by tmat** [comment]
Also, can be include `-restore` in `build.cmd`? It takes like 8 extra seconds on SSD.

**Comment by sharwell** [comment]
> Also, can be include -restore in build.cmd? It takes like 8 extra seconds on SSD.

I'd prefer not for now. It would add ~30% to my incremental build times.

File: eng/targets/GenerateCompilerExecutableBindingRedirects.targets (hunk 1 above)
Comment (sharwell) [reply eng/targets/GenerateCompilerExecutableBindingRedirects.targets]: Done thanks!

**Comment by tmat** [comment]
@sharwell You don't build from VS? I wouldn't expect the command line build to be used that often.

**Pull request merged**